Xml/src/Akka.Streams.Xml/Xml.cs

```diff
@@ -0,0 +1,707 @@
+                        break;
+
+                    // PROCESSING_INSTRUCTION
+                    case XmlNodeType.ProcessingInstruction:
+                        if (_parser.Depth == 0 && !_documentStarted)
+                        {
+                            // START_DOCUMENT
+                            _documentStarted = true;
+                            Push(_stage.Out, new StartDocument());
+                            _pendingEvent = new ProcessingInstruction(_parser.Name, _parser.Value);
+                        }
+                        else
+                            Push(_stage.Out, new ProcessingInstruction(_parser.Name, _parser.Value));
+                        break;
+
+                    // COMMENT
+                    case XmlNodeType.Comment:
+                        Push(_stage.Out, new Comment(_parser.Value));
+                        break;
+
+                    // CDATA
+                    case XmlNodeType.CDATA:
+                        Push(_stage.Out, new CData(_parser.Value));
+                        break;
+
+                    // Do not support DTD, SPACE, NAMESPACE, NOTATION_DECLARATION, ENTITY_DECLARATION, PROCESSING_INSTRUCTION
+                    // ATTRIBUTE is handled in START_ELEMENT implicitly
+                    // EVENT_INCOMPLETE is handled directly in AsyncXmlStream
+                    default:
+                        if (_feeder.Length - _feeder.Position < 7 && DataAvailable)
+                        {
+                            GetNextStreamBuffer();
+                            return;
+                        }
+                        AdvanceParser();
+                        return;
+                }
+            }
+        }
+        #endregion
+
+        public StreamingXmlParser(int bufferSize)
+        {
+            if (bufferSize < 64)
+                throw new ArgumentException($"Buffer size must be greater than 64 (was:{bufferSize})");
+
+            Shape = new FlowShape<ByteString, IParseEvent>(In, Out);
+            _bufferSize = bufferSize;
+        }
+
+        private readonly int _bufferSize;
+
+        protected override Attributes InitialAttributes { get; } = Attributes.CreateName("XmlParser");
+
+        public Inlet<ByteString> In { get; } = new Inlet<ByteString>("XMLParser.In");
+        public Outlet<IParseEvent> Out { get; } = new Outlet<IParseEvent>("XMLParser.out");
+
+        public override FlowShape<ByteString, IParseEvent> Shape { get; }
+
+        protected override GraphStageLogic CreateLogic(Attributes inheritedAttributes)
+            => new Logic(this, _bufferSize);
+
+        public override string ToString() => nameof(StreamingXmlParser);
+    }
+    #endregion
+
+    #region Coalesce
+    /// <summary>
+    /// Internal API. Use <see cref="Akka.Streams.Xml.Dsl.XmlParsing"/> instead.
+    /// </summary>
+    public sealed class Coalesce : GraphStage<FlowShape<IParseEvent, IParseEvent>>
```

Comment:
You can use `SimpleLinearGraphStage<IParseEvent>` in cases where  `in` and `out` type are the same